src/libraries/System.Text.Json/gen/PropertyGenerationSpec.cs

```diff
@@ -9,6 +9,13 @@ namespace System.Text.Json.SourceGeneration
     [DebuggerDisplay("Name={Name}, Type={TypeMetadata}")]
     internal sealed class PropertyGenerationSpec
     {
+        /// <summary>
+        /// The exact name specified in the source code. This might be different
+        /// from the <see cref="ClrName"/> because source code might be decorated
+        /// with '@' for reserved keywords, e.g. public string @event { get; set; }
+        /// </summary>
+        public string ExactNameSpecifiedInSourceCode { get; init; }
```

Comment:
I struggled a bit with the name for this. Happy to get alternatives as I don't think it's great. The presence of this new property also casts a bit of a shadow over `ClrName`; maybe we just need a `DecoratedName` and `UndecoratedName`?